dns_sd_avahi.c

```diff
@@ -83,6 +86,14 @@ static void avahi_process_resolved(struct dns_sd_discovery_data *ddata,
 	}
 	iio_mutex_unlock(ddata->lock);
 
+	ptr = ddata->addr_str + strlen(ddata->addr_str);
```

Comment:
codacy flags this:
strlen does not handle strings that are not \0-terminated; if given one it may perform an over-read

could be:
ptr = ddata->addr_str + strnlen(ddata->addr_str, DNS_SD_ADDRESS_STR_MAX);

Reply:
But the string is guaranteed to be \0-terminated, isn't it?

Reply:
It should be - as far as I can tell.

Since it's not protected from the app that's using it - a bad pointer somewhere else could over write things.

But you are right - it's low likelihood.